Mods/src/PuppiMod.cc

```diff
@@ -92,33 +101,43 @@ Int_t PuppiMod::GetEtaBin(const PFCandidate *cand)
 }
 
 //--------------------------------------------------------------------------------------------------
-Double_t PuppiMod::Chi2fromDZ(Double_t dz)
+Double_t
+PuppiMod::Chi2fromDZ(Double_t dz) const
 {
   Double_t probPV = ROOT::Math::normal_cdf_c(fabs(dz),fTrackUncertainty) * 2.0;
   Double_t probPU = 1 - probPV;
   Double_t chi = 0;
-  if (probPU == 1) 
+  if (probPU == 1)
     chi = 100;                                // This doesn't exactly match CMSSW, but I like this better
-  else 
+  else
     chi = TMath::ChisquareQuantile(probPU,1);
-  return pow(chi,2);
+  return chi * chi;
 }
 
 //--------------------------------------------------------------------------------------------------
-void PuppiMod::SlaveBegin()
+void
+PuppiMod::SlaveBegin()
 {
+  // Read the configuration file
```

Comment:
Moving check-and-abort to before a new operation. Aborting will take care of cleaning up allocated memory, but there is no need to do the check afterwards anyway.